crates/core/src/subscription/execution_unit.rs

```diff
@@ -206,8 +206,13 @@ impl ExecutionUnit {
 
     /// Evaluate this execution unit against the database using the json format.
     #[tracing::instrument(skip_all)]
-    pub fn eval_json(&self, db: &RelationalDB, tx: &Tx) -> Result<Option<TableUpdateJson>, DBError> {
-        let table_row_operations = Self::eval_query_expr(db, tx, &self.eval_plan, |row| {
+    pub fn eval_json(
+        &self,
+        ctx: &ExecutionContext,
+        db: &RelationalDB,
+        tx: &Tx,
```

Comment:
Because we want to write metrics at most once per transaction, it would probably be best to embed the ExecutionContext directly in the Tx. But that might require us to update a lot of function signatures.

Reply:
Ah I saw you opened #1030. Excellent!